visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -0,0 +1,263 @@
+<template>
+    <div class="relative flex t-date-picker">
+        <!-- Date Picker -->
+        <base-date-picker
+            read-only
+            show-clear-button
+            confirm
+            confirm-text="Apply"
+            v-model="date"
+            ref="datePicker"
+            :range="mode === 'range'"
+            :format="dateFormat"
+            :range-separator="seperator"
+            @confirm="handleConfirm"
+            @pick="setCustomPreset"
+            @open="fetchLayerData"
+        >
+            <div class="presets-dropown" slot="sidebar">
+                <button
+                    v-for="preset in presets[mode]"
+                    :key="preset.key"
+                    :style="{ color: preset.key === selectedPreset.key ? '#1284e7' : '#73879c' }"
+                    class="cursor-pointer"
+                    @click="handlePreset(preset)"
+                >
+                    {{ preset.label }}
+                </button>
+            </div>
+
+            <div slot="icon-clear"></div>
+            <div slot="icon-calendar">
+                <div class="flex items-center gap-1">
+                    <button>
+                        <reload-icon @click="handleReload" />
+                    </button>
+                    <button>
+                        <close-icon @click="handleClear" />
+                    </button>
```

Comment:
let's move the 'click' event to the buttons. That way the whole element is clickable, not just to the icon.